Sources/Sentry/SentryFramesTracker.m

```diff
@@ -71,6 +85,11 @@ - (void)resetFrames
     atomic_store_explicit(&_slowFrames, 0, SentryFramesMemoryOrder);
 
     self.previousFrameTimestamp = SentryPreviousFrameInitialValue;
+
+#    if SENTRY_TARGET_PROFILING_SUPPORTED
+    self.frameTimestamps = [[SentryMutableFrameTimestampInfo alloc]
+        initWithCapacity:SentryNumberOfFrameTimestampsToRetain];
+#    endif // SENTRY_TARGET_PROFILING_SUPPORTED
```

Comment:
`h`: We still need to check if `options.enableProfiling` is enabled, and only if it is, we should initialize this array. Otherwise, we init a huge array for nothing. Cause we use a shared instance and therefore getting access to the options could be a bit tricky, we could also init the array on the first call of `recordTimestampStart`.

Reply:
Actually, can't do this here, since this is initialized before the profiler begins. But as I mentioned, see https://github.com/getsentry/sentry-cocoa/pull/1910/files#diff-2bbc6b239c35033a741f7153082aaacbe74797706174425371a6538eded99429R154 for where we check if profiling is running before adding to this data structure.

Reply:
Didn't know that the SentryMutableFrameTimestampInfo doesn't allocate all the elements on `initWithCapacity`.

Reply:
Although `NSMutableArray initWithCapacity` doesn't allocate the whole array yet, the passed-in capacity is a hint to the NSMutable for how much memory it should allocate. I'm not 100% sure how the internals work, but it could be that we waste some memory for nothing.